android-DL4JImageRecognitionDemo.md

```diff
@@ -0,0 +1,382 @@
+---
+title: Using Deeplearning4J in Android Applications - demo 2
+layout: default
+---
+# Using Deeplearning4J in Android Applications
+
+Contents
+
+* [Setting the Dependencies](#head_link1)
+* [Training and loading the Mnist model in the Android project resources](#head_link2)
+* [Accessing the trained model using an AsyncTask](#head_link7)
+* [Handling images from user input](#head_link3)
+* [Updating the UI](#head_link5)
+* [Conclusion](#head_link6)
+
+## DL4JImageRecognitionDemo
+This example application uses a neural network trained on the standard MNIST dataset of 28x28 greyscale 0..255 pixel value images of hand drawn numbers 0..9. The application user interace allows the user to draw a number on the device screen which is then tested against the trained network. The output displays the most probable numeric values and the probability score. This tutorial will cover the use of a trained neural network in an Android Application, the handling of user generated images, and the output of the results to the UI from a background thread. For a detailed guide demonstrating how to train and save the neural networks used in this application, please see this DL4J quickstart [tutorial](https://deeplearning4j.org/quickstart). More information on general prerequisites for building DL4J Android Applications can be found [here](https://github.com/jrmerwin/Skymind-Android-Documentation/blob/master/Prereqs%20and%20Configuration%20for%20Android.md). 
+
+![](images/screen2.png)
+## <a name="head_link1">Setting the Dependencies</a>
+Deeplearning4J applications requires application specific dependencies in the build.gradle file. The Deeplearning library in turn depends on the libraries of ND4J and OpenBLAS, thus these must also be added to the dependencies declaration. Starting with Android Studio 3.0, annotationProcessors need to be defined as well, thus dependencies for either -x86 or -arm processors should be included, depending on your device, if you are working in Android Studio 3.0 or later. Note that both can be include without conflict as is done in the example app.
+```java
+	compile 'com.android.support:appcompat-v7:27.0.2'
+        compile 'com.android.support:design:27.0.2'
+        compile 'org.deeplearning4j:deeplearning4j-nn:0.9.1'
+        compile 'org.nd4j:nd4j-native:0.9.1'
+        compile 'org.nd4j:nd4j-native:0.9.1:android-x86'
+        compile 'org.nd4j:nd4j-native:0.9.1:android-arm'
+        compile 'org.bytedeco.javacpp-presets:systems-platform:1.4'
```

Comment:
We actually should just need `compile 'org.bytedeco:javacpp:1.4'` here.

Reply:
Yes you're right. I changed that dependency and it worked fine.